### PR TITLE
Fixes and unittests for factory api

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- increase code coverage
  
## Proposed changes
- added unittests for factory/api + small fixes

## Testing procedure
- standard system test

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: factory/api/apiResolverFactory.go [hunk 1]
```diff
@@ -408,7 +408,7 @@ func createScQueryElement(
 		queryVirtualMachineConfig := args.generalConfig.VirtualMachine.Querying.VirtualMachineConfig
 		esdtTransferParser, errParser := parsers.NewESDTTransferParser(args.coreComponents.InternalMarshalizer())
 		if errParser != nil {
-			return nil, err
+			return nil, errParser
```
Comment (iulianpascalau) [inline factory/api/apiResolverFactory.go:411]: :+1:

File: process/errors.go [hunk 2]
```diff
@@ -391,7 +391,7 @@ var ErrWrongTypeInMiniBlock = errors.New("type in miniblock is not correct for p
 var ErrNilTransactionCoordinator = errors.New("transaction coordinator is nil")
 
 // ErrNilUint64Converter signals that uint64converter is nil
-var ErrNilUint64Converter = errors.New("unit64converter is nil")
+var ErrNilUint64Converter = errors.New("uint64converter is nil")
```
Comment (iulianpascalau) [inline process/errors.go:394]: :+1:

File: factory/api/apiResolverFactory_test.go [hunk 3]
```diff
@@ -59,8 +104,357 @@ func TestCreateApiResolver(t *testing.T) {
...
+		apiResolver, err := api.CreateApiResolver(args)
+		require.True(t, strings.Contains(err.Error(), "VirtualMachine.Querying.NumConcurrentVms"))
+		require.True(t, check.IfNil(apiResolver))
+	})
+
+	failingStepsInstance := &failingSteps{}
+	failingArgs := createFailingMockArgs(t, failingStepsInstance)
+	// do not run these tests in parallel as they all use the same args
+	t.Run("DecodeAddresses fails causing createScQueryElement error should error", func(t *testing.T) {
+		failingStepsInstance.reset()
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply factory/api/apiResolverFactory_test.go:170]: Yup, we have kind of dead code on this factories as we always check for errors when we assemble components. They should not appear if we use the real components (such as the CoreComponentsHolder). However, in order to increase the code coverage, we need to test also those error branches. Those errors can happen only on 2 situations:
1. we do something wrong in the real CoreComponentsHolder and we end up returning a nil sub-component (the code will break in a lot of other parts)
2. we do another core component. This might happen if someone else re-uses this component in another repo. 
TL;DR, I would keep these tests.

**Review by iulianpascalau — APPROVED** [review]

**Review by schimih — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by schimih — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 
@@                            Log scanner                                  @@
##                            unittests_for_factory_api                    ##
================================================================================
- Known Warnings              16
- New Warnings                6
- Known Errors                1
- New Errors                  2
- Panics                      0
================================================================================
- block hash does not match                  37731
- wrong nonce in block                       12002
- miniblocks does not match                  0
- num miniblocks does not match              0
- miniblock hash does not match              0
- block bodies does not match                0
- receipts hash missmatch                    0
================================================================================
-  No jailed nodes on the thestnet
================================================================================

**Review by iulianpascalau — APPROVED** [review]
